program/llvm-toolchain-ppl.md

```diff
@@ -0,0 +1,252 @@
+
+In order to support the Software TG's primary goal of developing a thriving commercial ecosystem, only a basic implementation will be provided.
+By basic implementation we mean that:
+- the assembler/linker will support the instructions;
+- the compiler will have intrinsic (builtin) function support;
+- the compiler will have patterns to generate the instructions from C code in obvious circumstances.
+
+There will be no attempt to provide compiler optimizations.  The development of optimizations will be under the responsibility of:
+- software companies which can rely on the amount of work still needing to be done to drive their businesses; and of
+- HW and system integration companies which can use specific optimizations targeted at their products as market differentiators.
+
+In order to support the Software TG secondary goal of upstreaming all open source tool developments:
+- all Clang/LLVM development will be kept compliant with the LLVM Foundation coding and quality assurance standards;
+- the implementation will follow the upstream tool design and coding conventions; and
+- any vendor-specific modifications or additions will be duly isolated into extensions, so as to maintain a fully functional common open source code base.
+
+Finished LLVM-related work that is not vendor-restricted will be contributed to the LLVM community and maintained upstream.
+
+### Nature of the development
+
+This project requires the modification of a set of existing public open source code bases maintained as projects of the LLVM Foundation and the REMS project.  Therefore, the processes used within OpenHW will reflect the processes of those upstream projects.
+
+The code base of LLVM is of substantial size:
+
+- Clang: 1.75 MLOC C++;
+- LLVM: 2.1 MLOC C++.
+
+The LLVM code base includes substantial regression test suites (1.5 MLOC including `libcxx` C++ library tests), and success with these test suites is a pre-requisite of upstream acceptance of any patch.
+
+Until CORE-V related contributions are accepted upstream, code will be developed in narrow mirrors of upstream repositories, featuring a single branch based on upstream top of tree.  This will be used as the basis of the patches to be submitted for upstreaming.
+
+<!--- Partners may maintain local backport branches for corrections to official releases used in their internal HW development flows. --->
+
+Once CORE-V contributions are accepted upstream, common CORE-V specific code for LLVM will be developed and maintained in upstream `llvm.org` repositories.  However, dedicated extensions may continue to be developed in member-owned repositories tracking the upstream repository.
+
+### Summary timeline
+
+- Preliminary Project Launch (PPL, this document): TBD
+  - includes project plan
+
+- Project Launch (PL): TBD
+- First OHG release (CVA6 64/32 bits, HW verification equivalence to GCC): TBD
+- Upstreaming of first release: 2021 H1, targeting LLVM 13.0.
```

Comment:
OHG support for CV32E40P incremental, commenced 19 Nov 2020, aim to have assembler/builtin support by early Q1/2021. Support for compiler patterns and codegen will await additional resource commitment.